### DISCUSSION TITLE
Check-in and check-out (input / output management).

### DISCUSSION THREAD

**Discussion by meteos77** [textbrowser/biblioteq#149] [Q&A]
today received about 400 documents
received 2 .pan files (UNIMARC-ISO5426), there were duplicates in the files

after manual inventory 82 books are missing in the files provided
the BDP asks for a list of access numbers

how would you manage this problem in general and with BiblioteQ do you have any ideas ??

**Comment by textbrowser** [comment]
I would need to know what you'd like to do with the 82 books that are missing.

**Reply by meteos77** [reply]
for the moment, they are put aside while waiting for the unimarc-iso file (provided by the Departmental Library) which allows to enter them in the software and to manage them by the shower.

but the way to manage them is not optimal

I "stupidly" scanned the documents to know if they were well integrated in the software, they were not there (because not in the files) and I put them aside.

After contacting the Departmental Library, they asked me for the list (accession number). The current software did not retain my scans which indicated that the books were not in the database, so I have to make a text file with the accession number:
=> Result rescan the 82 documents (work done twice).

I am looking for an idea to solve this problem in the future

it's a kind of inventory in the form of a text file that indicates the accession_number found or not found : there I have no trace

**Comment by meteos77** [comment]
The opposite phenomenon exists
the unimarc-iso list contains documents that we don't have physically (human error of the Central Library)

as a result last year we had to pay 800€ of books to the Bibliothèque Départementale (lost books invoiced... knowing that our budget is less than 600€) 
this year I decided to monitor the entries / exits very precisely.

**Comment by meteos77** [comment]
after reflection I submit my proposals ; I would prefer the 2nd one but I don't know if it's a mountain and I am waiting for your own solution to manage this feature
[2021-12-19-Reflections on document entry.txt](https://github.com/textbrowser/biblioteq/files/7740541/2021-12-19-Reflections.on.document.entry.txt)

for me in french :
[2021-12-19-en_français_Réflexions-Processus-Entree-Documents.txt](https://github.com/textbrowser/biblioteq/files/7740544/2021-12-19-en_francais_Reflexions-Processus-Entree-Documents.txt)

**Reply by textbrowser** [reply]
> after reflection I submit my proposals ; I would prefer the 2nd one but I don't know if it's a mountain and I am waiting for your own solution to manage this feature [2021-12-19-Reflections on document entry.txt](https://github.com/textbrowser/biblioteq/files/7740541/2021-12-19-Reflections.on.document.entry.txt)
> 
> for me in french : [2021-12-19-en_français_Réflexions-Processus-Entree-Documents.txt](https://github.com/textbrowser/biblioteq/files/7740544/2021-12-19-en_francais_Reflexions-Processus-Entree-Documents.txt)

Solution #2 requires that the imported items undergo the reservation process. This would duplicate the reservation process.

**Reply by meteos77** [reply]
Importing and placing documents in borrowing position through the import process with a fixed borrowing account is too difficult?

**Reply by meteos77** [reply]
I thought that the solution N°2 was easier to implement than N°1 ?

**Reply by textbrowser** [reply]
Neither are pleasant. However, this is for your special import. Reserving an item involves more than some database inserts. I will look. :P

**Reply by meteos77** [reply]
do you have a proposal N° 3 ?

**Reply by meteos77** [reply]
you tell me if I can help you on the special import

**Comment by textbrowser** [comment]
The specialized import includes a corrective process, yes?

**Reply by meteos77** [reply]
what you call specialized import includes what for you?
I do not understand well corrective process?

**Reply by textbrowser** [reply]
The import we've been discussing.

Solution N2 is a corrective process, meaning that it places some of the items in special states. A traditional import doesn't correct entries. If entries do not include the required information, it either discards them or invents values if possible. So both solution N1 and N2 require special logic to put the records in some special states.

**Reply by meteos77** [reply]
solution N° 1, if the import file contains the field 995 fill a field in the database that indicates NO 
AVAILABLE
a bit like the check mark "book status read   except that it blocks the loans

disadvantage : compared to the 2nd solution, you need an additional interface to switch the field NOT AVAILABLE into AVAILABLE
